crates/loader/src/validation.rs

```diff
@@ -1,19 +1,37 @@
-use anyhow::{anyhow, Result};
-
-const ONLY_ALLOWED_KV_STORE: &str = "default";
+use anyhow::{ensure, Context, Result};
 
 pub(crate) fn validate_key_value_stores(key_value_stores: &Option<Vec<String>>) -> Result<()> {
-    match key_value_stores
-        .iter()
-        .flatten()
-        .find(|id| *id != ONLY_ALLOWED_KV_STORE)
-    {
-        None => Ok(()),
-        Some(invalid) => {
-            let err = anyhow!("Invalid key-value store '{invalid}'. This version of Spin supports only the '{ONLY_ALLOWED_KV_STORE}' store.");
-            Err(err)
-        }
+    for store in key_value_stores.iter().flatten() {
+        validate_component_like_label(store)
```

Comment:
Yeah that would be quite awkward to manage here. It could make sense to add a new e.g. `DynamicHostComponent::validate_app(&self, app: &App) -> Result<()>` method that could be plumbed through to validate things like that, called from `spin-trigger`.

Reply:
https://github.com/fermyon/spin/issues/1371